File: frontend/src/app/models/test/SimulationFaultBatch.ts
import { AlgorithmType } from "../algorithm/AlgorithmType";
import { Robot } from "../algorithm/Robot";
import { Graph } from "../graph/Graph";
import { GraphType } from "../graph/GraphType";

export interface SimulationFaultBatch {
  graphType: GraphType;
  graph: Graph;
  algorithmType: AlgorithmType;
  robots: Robot[];
  numOfTests: number;

  // Fault
  faults: number;
  probability: number;
}

File: backend/src/main/java/hu/elte/ik/backend/model/test/SimulationFaultBatch.java
package hu.elte.ik.backend.model.test;

import hu.elte.ik.backend.model.algorithm.Robot;
import hu.elte.ik.backend.model.graph.Edge;
import hu.elte.ik.backend.model.graph.Node;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class SimulationFaultBatch<
  NodeType extends Node, EdgeType extends Edge, RobotType extends Robot
>
  extends SimulationBatch<NodeType, EdgeType, RobotType> {

  private Integer faults;
  private Double probability;
}
